### PR TITLE
Separate location of analysis cache if we're cross compiling scala.

### LINKED ISSUE #1267: sbt deleting cross-build classes directory without provocation
## steps

```
$ cat project/build.properties
sbt.version=0.13.2
$ cat build.sbt               
name := "foo"

scalaVersion := "2.10.4"

crossScalaVersions := List("2.10.4", "2.11.0-RC4")
$ cat foo.scala 
package foo

object Foo extends App {
  println("yay")
}
```

```
% ls -1d target/*/classes
target/scala-2.10/classes
% sbt ++2.11.0-RC4 compile
[success] Total time: 2 s, completed Apr 16, 2014 7:03:22 PM
% ls -1d target/*/classes
target/scala-2.11.0-RC4/classes
% sbt +compile
[info] Setting version to 2.10.4
[success] Total time: 2 s, completed Apr 16, 2014 7:06:34 PM
[info] Setting version to 2.11.0-RC4
[success] Total time: 2 s, completed Apr 16, 2014 7:06:36 PM
[info] Setting version to 2.10.4
% ls -1d target/*/classes
target/scala-2.11.0-RC4/classes
% sbt ++2.10.4 compile
[info] Setting version to 2.10.4
[success] Total time: 2 s, completed Apr 16, 2014 7:06:57 PM
% ls -1d target/*/classes
target/scala-2.10/classes
%
```
## problem

sbt deletes unused target/*/classes directory on cross building.
## expectation

sbt does not delete target/*/classes directory from prior cross building.

### PR DESCRIPTION
**Pull request by jsuereth**
- Add task to determine file-name of analysis cache
- If crossPaths := true, then add scala binary version to the analysis cache name.

This makes it possible to leverage incremental compilation with the `+` command.

Fixes #1267 

Review by @gkossakowski  cc> @eed3si9n

### CODE REVIEW

File: main/src/main/scala/sbt/Keys.scala [hunk 1]
```diff
@@ -142,6 +142,7 @@ object Keys
 	val consoleProject = TaskKey[Unit]("console-project", "Starts the Scala interpreter with the sbt and the build definition on the classpath and useful imports.", AMinusTask)
 	val compile = TaskKey[Analysis]("compile", "Compiles sources.", APlusTask)
 	val compilers = TaskKey[Compiler.Compilers]("compilers", "Defines the Scala and Java compilers to use for compilation.", DTask)
+	val compileAnalysisFile = TaskKey[String]("compileAnalysisFile", "Defines the filename used to store the incremental compiler analysis file (inside the streams cacheDirectory).", DTask)
```
Comment (eed3si9n) [inline main/src/main/scala/sbt/Keys.scala]: If the key is postfixed with the name "File", I'd want `File`. `analysisFileName` maybe?

Comment (jsuereth) [inline main/src/main/scala/sbt/Keys.scala]: Can do

**Comment by jsuereth** [comment]
@eed3si9n Ok, fixed the name.

**Comment by gkossakowski** [comment]
LGTM

**Pull request merged**